eureka-client/src/main/java/com/netflix/discovery/InstanceInfoReplicator.java

```diff
@@ -68,10 +69,27 @@ public void start(int initialDelayMs) {
     }
 
     public void stop() {
-        scheduler.shutdownNow();
+        Future latestPeriodic = scheduledPeriodicRef.get();
+        if (latestPeriodic != null && !latestPeriodic.isDone()) {
+            logger.info("Canceling the latest scheduled before stop");
+            boolean result = latestPeriodic.cancel(false);
```

Comment:
You shouldn't need to cancel the future here, shutting down the scheduler should handle that as the future is scheduled on that.